### PR TITLE
[Pytorch] Add ignore_raw_node option to GraphPickler

### PR DESCRIPTION
**Pull request by nakuliyer**
Summary:
The Graph Pickler raises "Unexpected raw Node during pickling" when it encounters a raw `torch.fx.Node` during serialization. This occurs in some backward graphs (e.g. IDXCVR model__27_backward_7) where metadata contains Node references that haven't been converted to `_NodePickleData`.

This diff adds an `ignore_raw_node` option to `GraphPickler.Options` that, when enabled, silently pickles unexpected raw Nodes as `None` instead of raising an `AssertionError`. This allows graph serialization to finish even when graphs contain these edge-case references.

Test Plan: buck test fbcode//mtia/accuracy/training/tests:test_graph_serialize; arc lint — no new issues

Reviewed By: aorenste

Differential Revision: D95510476

### CODE REVIEW

File: torch/fx/_graph_pickler.py [hunk 1]
```diff
@@ -143,6 +150,8 @@ def reducer_override(
         else:
             # We should never get a raw Node!
             if isinstance(obj, torch.fx.Node):
+                if self.options.ignore_raw_node:
+                    return (_unpickle_as_none, ())
                 raise AssertionError("Unexpected raw Node during pickling")
```
[the remark this answers is not in the document]
Comment (nakuliyer) [reply torch/fx/_graph_pickler.py:155]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/pytorch/pytorch/pull/176939#discussion_r2908263209)

**Review by aorenste — APPROVED** [review]
Needs lint, but LGTM
